src/mainboard/sifive/hifive/src/main.rs

```diff
@@ -92,40 +91,40 @@ pub extern "C" fn _start_boot_hart(_hart_id: usize, fdt_address: usize) -> ! {
 
     let w = &mut print::WriteTo::new(uart0);
 
-    write!(w, "## ROM Device Tree\r\n").unwrap();
+    writeln!(w, "## ROM Device Tree").unwrap();
```

Comment:
Usually when I see a `\r`, I pause because it's usually there very deliberately. Do you know why it was there? Will `writeln!` preserve it?

Reply:
No idea why it was here, but it seems like the code in this file is randomly using \n and \r\n (e.g. 3 lines below in line 98 it uses just `\n`). Because this is mostly for debugging, and the CI tests pass, I would like to suggest to just drop \r and use writeln everywhere in this file. If at any point in time we find that \r is actually needed, then we can easily change it back and we will have clear reason that we could document at the same time. WDYT?

Reply:
does writeln send a \r or does it assume it? If it does not send \r then that's a problem.

Reply:
"On all platforms, the newline is the LINE FEED character (\n/U+000A) alone (no additional CARRIAGE RETURN (\r/U+000D)." —https://doc.rust-lang.org/std/macro.writeln.html

Ron, can you clarify why `\r` is necessary? It appears that some of the old code had it and some didn't.

Reply:
Yes `\r\n` is necessary.

`\r` is carriage return; goes to the start of the current line. `\n` is newline; goes down one line without shifting horizontally.

Something like "hello\nworld" would print like this:

```
hello
     world
```

You might wonder why you've never had to use `\r` before.

This is because application-level code in Linux is using a "cooked" TTY. In the cooked mode, `\n` is automatically replaced by `\r\n` before sending out to serial. Different operating systems do this differently (like Windows expects all the files to contain \r\n).

You can experience raw uncooked power in Linux with:

```
~$ stty raw
~$ printf "hello\nworld"
hello
     world~$ 
```

Also, running oreboot in qemu with `-nographic` is misleading because it piggybacks off of the surrounding TTY in Linux. But, I assure you when you run on hardware and connect a serial device, if you only use `\n` the newlines will look weird like in the first example.

Reply:
Thanks, Ryan!

Would it make sense to write our own `writern!` macro that does `\r\n`? Or can we add a custom linting rule to squawk at `writeln!` and `write!` with a bare `\n`? (No idea how one would best add a custom linting rule.)

Reply:
Thank you for the detailed explanation Ryan! I reverted the changes to `write!` and disabled the check in clippy. I also created #321 to focus on this (if possible, I would like to separate the simple clippy fixes from fixing the write! macros).